nncf/dynamic_graph/context.py

```diff
@@ -516,11 +517,16 @@ def reset_graph(self):
 @contextmanager
 def no_nncf_trace():
     ctx = get_current_context()
-    if ctx is not None:
-        ctx.disable_tracing()
-    yield
-    if ctx is not None:
-        ctx.enable_tracing()
+    if ctx:
+        if ctx.is_tracing:
+            ctx.disable_tracing()
+            yield
+            if ctx:
+                ctx.enable_tracing()
+        else:
+            yield
+    else:
+        yield
```

Comment:
This code looks worse that the one above.
1) Avoid nestings if possible, use compound conditions.
2) Do not do `if a:` when you actually try to test if `a` name points to None. Do `if a is not None`.
3) What problem does this solve? Such explanations should be present in the commit message.

Reply:
1. Updated
2. Updated
3. Added a description. I hope, now it is more clear

Reply:
Please also add a simple test for this logic. Sorry I did not mention it in the previous message.

Reply:
Add a test to `test_graph_building.py`